src/main/java/com/globallogic/amcr/repository/attachmentcomponent/AttachmentDao.java

```diff
@@ -0,0 +1,25 @@
+package com.globallogic.amcr.repository.attachmentcomponent;
+
+import com.globallogic.amcr.model.attachmentcomponent.Attachment;
+import com.globallogic.amcr.model.attachmentcomponent.AttachmentMetadata;
+import com.globallogic.amcr.model.attachmentcomponent.AttachmentResponse;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface AttachmentDao {
+
+    byte[] getBinary(UUID id);
+
+    Attachment update(UUID id, byte[] content, Attachment oldAttachment);
+
+    void delete(UUID id);
+
+    Attachment save(Attachment attachment, UUID id);
+
+    Attachment get(UUID id);
```

Comment:
So I have different GET endpoints. 

1. getMetadata (localhost/attachment/metadata/attachmentId)- returns a JSON with id, name, size, type and CRC.
2. getBytes (localhost/attachments/content/id) - returns the bytes of the attachment
3. getAllAttachmentsMetadata (localhost/attachments/) - similar to getMetadata, except it returns the 20 most recent ones at once (typically I don't see the need for it just yet, but thought I'd add it just in case it could be relevant in the future)
4. getMedia (localhost/atatchments/id) - returns the attachments itself, a picture or a PDF depending on what you have uploaded.

I don't think it's the same endpoint as each endpoint returns different things. I am not sure and I don't know how to change this.